common-arcana.lic

```diff
@@ -72,39 +72,48 @@ module DRCA
     end
   end
 
-  def start_barb_abilities(skills, settings)
-    skills
+  def start_barb_abilities(abilities, settings)
+    # Note, you must know Power meditation or Powermonger mastery
+    # for your active abilities to be detected by DRSpells.
+    abilities
       .reject { |name| DRSpells.active_spells[name] }
-      .each do |name|
-        activate_barb_buff?(name)
-        waitrt?
-        pause settings.meditation_pause_timer.to_i if get_data('spells').barb_abilities[name]['type'].eql?('meditation')
-        DRC.fix_standing
-      end
+      .each { |name| activate_barb_buff?(name, settings) }
   end
 
-  def activate_barb_buff?(name)
+  def activate_barb_buff?(name, settings = nil)
+    activated = false
     ability_data = get_data('spells').barb_abilities[name]
     case DRC.bput(ability_data['start_command'], ability_data['activated_message'], 'But you are already', 'Your inner fire lacks', 'find yourself lacking the inner fire', 'You should stand', 'You must be sitting', 'You must be unengaged', 'While swimming?')
     when 'You must be unengaged'
       DRC.retreat
-      activate_barb_buff?(name)
+      activated = activate_barb_buff?(name, settings)
     when 'You must be sitting'
       DRC.retreat
       case DRC.bput('sit', 'You sit', 'You are already', 'You rise', 'While swimming?')
       when 'While swimming?'
-        false # can't sit here, water is too deep
+        activated = false # can't sit here, water is too deep
       else
-        activate_barb_buff?(name)
+        activated = activate_barb_buff?(name, settings)
       end
     when 'You should stand'
       DRC.fix_standing
-      activate_barb_buff?(name)
-    when ability_data['activated_message']
-      true
+      activated = activate_barb_buff?(name, settings)
+    when /#{ability_data['activated_message']}/
+      # Pause at least for the preferred amount of time
+      # to let the meditation take effect else it may fail.
+      # Note, we use bracket notation to reference settings here
+      # because this method may be invoked with either JSON or an OpenStruct.
+      if ability_data['type'].eql?('meditation') && settings['meditation_pause_timer']
+        pause settings['meditation_pause_timer'].to_i
+      end
+      # Wait for any remaining RT before proceeding
+      waitrt?
+      activated = true
     else
-      false
+      activated = false
     end
+    DRC.fix_standing
+    activated
```

Comment:
After attempting to activate an ability, always try to stand back up in case you had to sit down